common/credentials-interface/src/lib.rs

```diff
@@ -74,17 +90,168 @@ impl Display for CredentialType {
+            return Err(CompactEcashError::Deserialization(
+                "Invalid byte array for EcashCredential deserialization".to_string(),
+            ));
+        }
+        let payment = Payment::try_from(&raw[index..index + payment_len])?;
+        index += payment_len;
+
+        if raw[index..].len() < 72 + 8 + 8 + 8 + 4 {
+            return Err(CompactEcashError::Deserialization(
+                "Invalid byte array for EcashCredential deserialization".to_string(),
+            ));
+        }
+
+        let pay_info = PayInfo {
+            //SAFETY : casting a slice of lenght 72 into an array of size 72
+            pay_info_bytes: raw[index..index + 72].try_into().unwrap(),
+        };
+        index += 72;
+
+        //SAFETY : casting a slice of lenght 8 into an array of size 8
+        let spend_date = u64::from_be_bytes(raw[index..index + 8].try_into().unwrap());
+        index += 8;
+
+        //SAFETY : casting a slice of lenght 8 into an array of size 8
+        let value = u64::from_be_bytes(raw[index..index + 8].try_into().unwrap());
+        index += 8;
+
+        //SAFETY : casting a slice of lenght 4 into an array of size 4
+        let typ_len = u32::from_be_bytes(raw[index..index + 4].try_into().unwrap()) as usize;
+        index += 4;
+
+        if raw[index..].len() != typ_len + 8 {
+            return Err(CompactEcashError::Deserialization(
+                "Invalid byte array for EcashCredential deserialization".to_string(),
+            ));
+        }
+
+        let raw_typ = String::from_utf8(raw[index..index + typ_len].to_vec()).map_err(|_| {
+            CompactEcashError::Deserialization("Failed to deserialize type".to_string())
+        })?;
+        let typ = raw_typ.parse().map_err(|_| {
+            CompactEcashError::Deserialization("Failed to deserialize type".to_string())
+        })?;
+        index += typ_len;
+
+        //SAFETY : casting a slice of lenght 8 into an array of size 8
+        let epoch_id = u64::from_be_bytes(raw[index..index + 8].try_into().unwrap());
+
+        Ok(CredentialSpendingData {
+            payment,
+            pay_info,
+            spend_date,
+            value,
+            typ,
+            epoch_id,
+        })
+    }
+}
+
+impl Bytable for CredentialSpendingData {
+    fn to_byte_vec(&self) -> Vec<u8> {
+        self.to_bytes()
+    }
+
+    fn try_from_byte_slice(slice: &[u8]) -> Result<Self, CompactEcashError> {
+        Self::try_from_bytes(slice)
+    }
+}
+
+impl Base58 for CredentialSpendingData {}
+
+pub use nym_coconut::{
+    hash_to_scalar, keygen as coconut_keygen, prove_bandwidth_credential, verify_credential,
+    Attribute, Base58 as CoconutBase58, BlindedSerialNumber, CoconutError,
+    Parameters as CoconutParameters, Signature as CoconutSignature, VerificationKey,
+    VerifyCredentialRequest,
+};
+
+//SW NOTE: for coconut compatibility
+pub fn to_coconut(verification_key: &VerificationKeyAuth) -> Result<VerificationKey, CoconutError> {
```

Comment:
do we need it?

Reply:
Yes, for backward compatibility. It could a `From` fn, but it would imply either one of the lib to depend on the other, and my plan was to phase out coconut

Reply:
TO FIX : scrape backwards compatibility

Reply:
Fixed in https://github.com/nymtech/nym/commit/c6c3b2f644b6c608f9f90ea3bd75aded0d45d01e